### PR TITLE
update correct field

### PR DESCRIPTION
**Pull request by DylanDylann**
### Fixed Issues

$ https://github.com/Expensify/App/issues/54493 
https://github.com/Expensify/App/issues/54495

PROPOSAL:

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Create a workspace and invite a member
2. Make the member to be an admin
3. Sign in with the secondary admin account on a different window
4. As the secondary admin go to Workspace > Profile > Plan (Keep the RHP open)
5. As the owner change the plan type
6. As the secondary admin, verify that the selected option is updated

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I used [JaimeGPT](https://chatgpt.com/g/g-2dgOQl5VM-english-to-spanish-translator-aka-jaimegpt) to get English > Spanish translation. I then posted it in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I added [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) for any new feature or bug fix in this PR to help automatically prevent regressions in this user flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/0ba2b12b-3e3d-4358-b4d5-61e989555732

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/0e225b8b-f306-434b-b14e-b43b800c6673

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/ea5fa266-d088-4f92-8119-aa8366fde855

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/9dd64054-ae84-4241-95da-e50481a7accf

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/3290f0c8-8e84-4f1d-9034-89112fe09d6c

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/8d7450eb-e61e-45cb-aadb-332849a7844a

</details>

### CODE REVIEW

**Comment by DylanDylann** [comment]
@jayeshmangwani Could you review this PR?

**Comment by jayeshmangwani** [comment]
@DylanDylann I'll

**Comment by DylanDylann** [comment]
You already saw that in the original PR but I missed committing the suggestion for the failure data :(

**Comment by jayeshmangwani** [comment]
Haha, we sometimes miss things. I’m reviewing PR in 5 minutes.

**Comment by jayeshmangwani** [comment]
@DylanDylann Upon pressing downgrade, nothing happens for me—it just reloads the page. Are you facing the same issue?

https://github.com/user-attachments/assets/da1f2baa-5b48-4911-b2f7-06d3aaf14d6c

**Comment by DylanDylann** [comment]
As mentioned [here](https://github.com/Expensify/App/issues/54493#issuecomment-2560880974), It is a BE bug. currently, the BE only allows the owner of policy downgrade

**Comment by jayeshmangwani** [comment]
@DylanDylann , it seems we need to adjust the permissions here. We should display the Plan Type if the user is the owner of the workspace, rather than the Admin.
https://github.com/Expensify/App/blob/8b7096fc4c4d522407a6f51b4fe0fd3543e5feb2/src/pages/workspace/WorkspaceProfilePage.tsx#L331-L334

**Comment by DylanDylann** [comment]
I think admin should have permission to downgrade and upgrade the workspace as design docs

**Comment by jayeshmangwani** [comment]
Okay then BE needs to be adjusted as I am getting BE error

```js

{
    "code": 666,
    "jsonCode": 666,
    "type": "Expensify\\Libs\\Error\\ExpError",
    "UUID": "4d7b2da5-d585-403a-aee7-793621b49e49",
    "message": "Only policy owner can downgrade",
    "title": "",
    "data": [],
    "htmlMessage": "",
    "onyxData": [],
    "requestID": "8f6fa64f085df29a-BOM"
}

```

**Comment by DylanDylann** [comment]
@jayeshmangwani I have added a new commit to solve [this bug](https://github.com/Expensify/App/issues/54495), too

**Comment by jayeshmangwani** [comment]
Thanks, testing that issue too

**Comment by DylanDylann** [comment]
@jayeshmangwani There is a minor problem. The App already updated the selected option when another device updated it. But the background focus still be outdated see the below video

https://github.com/user-attachments/assets/85e09c99-9ffc-4f7b-9e9e-8ae73bd9678e

But this is a bug in SelectionList that will happen to all places where we use SelectionList. Now, we already have [a PR](https://github.com/Expensify/App/pull/54457) to address it separately. I think we can ignore it here

**Comment by jayeshmangwani** [comment]
>  I think we can ignore it here

Sounds good to me

**Comment by jayeshmangwani** [comment]
@DylanDylann PR changes looks good, but can we please update the QA Steps to include the testing steps instead of doc link? It seems the tester doesn’t have access to the document , as they’ve commented on this PR https://github.com/Expensify/App/pull/54356#issuecomment-2560320808. Please update that PR with QA Steps as well.

**Comment by DylanDylann** [comment]
@jayeshmangwani Done. I updated test step for https://github.com/Expensify/App/issues/54495 only

**Comment by DylanDylann** [comment]
To test https://github.com/Expensify/App/issues/54493, I think we need to wait for the BE change

**Comment by jayeshmangwani** [comment]
>  I think we need to wait for the BE change

Yes we can wait, I think we can update this steps once BE will be fixed

File: src/pages/workspace/WorkspaceProfilePlanTypePage.tsx [hunk 1]
```diff
@@ -49,6 +49,10 @@ function WorkspaceProfilePlanTypePage({policy}: WithPolicyProps) {
         OpenWorkspacePlanPage(policyID);
     }, [policyID]);
 
+    useEffect(() => {
+        setCurrentPlan(policy?.type);
+    }, [policy?.type]);
```
Comment (jayeshmangwani) [inline src/pages/workspace/WorkspaceProfilePlanTypePage.tsx:54]: We need to update the active selection UI too, it doesn't looks good upon update

Screenshot 2024-12-24 at 3 43 45 PM

Comment (DylanDylann) [reply src/pages/workspace/WorkspaceProfilePlanTypePage.tsx:54]: @jayeshmangwani

> We need to update the active selection UI too,

What do you mean? It is mentioned [here](https://github.com/Expensify/App/pull/54501#issuecomment-2560929327)

Comment (jayeshmangwani) [reply src/pages/workspace/WorkspaceProfilePlanTypePage.tsx:54]: Please ignore this; I forgot about [that](https://github.com/Expensify/App/pull/54501#issuecomment-2560929327).

**Comment by jayeshmangwani** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] For any bug fix or new feature in this PR, I verified that sufficient [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) are included to prevent regressions in this flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/dfd2f410-b3ea-4098-81d7-f60ac308a899




</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/846f3c92-0ca0-47b3-865a-b4e48542f42a




</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/e2032498-918d-4677-8ebe-0f79c20e4c0e




</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/2318b2d9-a7db-45e6-8e83-80f7355af774




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/1b5e234c-74ba-4fe1-8ec6-46a24599149c




</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/26bddce3-a75d-4b38-ab23-76b184ba8a28




</details>

**Review by jayeshmangwani — APPROVED** [review]

**Review by techievivek — APPROVED** [review]

**Pull request merged**